lib/Fencer/Rules.hs

```diff
@@ -65,14 +65,28 @@ loadRulesFromDirectory
     let filteredFiles = if ignoreDotFiles
         then filter (not . isDotFile) files
         else files
-    (errs, rules) <- partitionEithers <$> mapM loadFile filteredFiles
-    pure $ if (null @[] errs) then Right rules else Left errs
+    (errs, mRules) <- partitionEithers <$> mapM loadFile filteredFiles
+    pure $ if (null @[] errs) then Right (catMaybes mRules) else Left errs
   where
-    loadFile :: FilePath -> IO (Either LoadRulesError DomainDefinition)
+    loadFile :: FilePath -> IO (Either LoadRulesError (Maybe DomainDefinition))
     loadFile file = catch
-      ((mapLeft (LoadRulesParseError file)) <$> Yaml.decodeFileEither @DomainDefinition file)
+      (parseErrorHandle file <$> Yaml.decodeFileEither @DomainDefinition file)
       (pure . Left . LoadRulesIOError)
 
+    -- | Handle a special case when the input file cannot be read due
+    -- to file permissions by returning Nothing on the Right.
+    parseErrorHandle
+      :: FilePath
+      -> Either Yaml.ParseException DomainDefinition
+         ----------------------------------------------
+      -> Either LoadRulesError (Maybe DomainDefinition)
+    parseErrorHandle _    (Right def)  = Right $ Just def
+    parseErrorHandle file (Left parEx) = case parEx of
+      Yaml.InvalidYaml (Just (Yaml.YamlException _)) ->
+        Right Nothing
```

Comment:
The comment says "when the input file cannot be read due to file permissions", but here you check for arbitrary YAML exceptions?

Reply:
@neongreen was faster than me on this.